login.js

```diff
@@ -0,0 +1,32 @@
+const passwordElem = document.getElementById("selected-user-password");
+const loginElem = document.querySelector(".login-btn");
+const selectElem = document.getElementById("select");
+
+let registeredUsers = JSON.parse(localStorage.getItem("MB_USER_ACCOUNTS"));
+for (let i = 0; i < registeredUsers.length; i++) {
+  let optionElem = document.createElement("option");
+  let text = document.createTextNode(registeredUsers[i].accountName);
+  optionElem.appendChild(text);
+  optionElem.setAttribute("value", registeredUsers[i].accountNumber);
+  selectElem.insertBefore(optionElem, selectElem.lastChild);
+}
+
+// function for logging in
+function login() {
+  localStorage.setItem(
+    "MB_LOGGEDIN_USER_ACCOUNT_NUMBER",
+    JSON.stringify(selectElem.value)
+  );
+  if (localStorage.getItem("MB_LOGGEDIN_USER_ACCOUNT_NUMBER") !== null) {
+    location.href = "transaction.html";
+  }
+  //to check user object with the stored account number
+  let accountUserDetails = registeredUsers.find(
```

Comment:
Suggested:

```
const accountUserDetails = registeredUsers.find(
    (user) =>
      user.accountNumber ===
      localStorage.getItem("MB_LOGGEDIN_USER_ACCOUNT_NUMBER")
  );
```

Pay attention to the changes I made:
1. Used `const` keyword instead of `let`, because the variable is not being re-assigned.
2. In the `find` method, I renamed each instance of `registeredUsers` from `acctNo` to `user`, because each element (object) in the `registeredUsers` array stores multiple user data (name, account number, pin, etc) and not just account number, so using `acctNo` is misleading.
3. No need to `JSON.parse ` the account number, since it's stored as a string and we need to use it as a string. You only ever need to `JSON.parse` array and objects strings i.e. arrays and objects that must have been converted to strings using `JSON.stringify`.